pkg/cmd/attestation/verification/sigstore.go

```diff
@@ -51,108 +55,98 @@ var ErrNoAttestationsVerified = errors.New("no attestations were verified")
 	if leafCert == nil {
-		return nil, "", fmt.Errorf("leaf cert not found")
+		return "", fmt.Errorf("leaf cert not found")
 	}
 	if len(leafCert.Issuer.Organization) != 1 {
-		return nil, "", fmt.Errorf("expected the leaf certificate issuer to only have one organization")
+		return "", fmt.Errorf("expected the leaf certificate issuer to only have one organization")
 	}
-	issuer := leafCert.Issuer.Organization[0]
+	return leafCert.Issuer.Organization[0], nil
+}
 
-	if v.config.TrustedRoot != "" {
-		customTrustRoots, err := os.ReadFile(v.config.TrustedRoot)
+func (v *LiveSigstoreVerifier) chooseVerifier(issuer string) (*verify.SignedEntityVerifier, error) {
+	// if no custom trusted root is set, attempt to create a Public Good or
+	// GitHub Sigstore verifier
+	if v.TrustedRoot == "" {
+		switch issuer {
+		case PublicGoodIssuerOrg:
+			if v.NoPublicGood {
+				return nil, fmt.Errorf("detected public good instance but requested verification without public good instance")
+			}
+			return newPublicGoodVerifier()
+		case GitHubIssuerOrg:
+			return newGitHubVerifier(v.TrustDomain)
+		default:
+			return nil, fmt.Errorf("leaf certificate issuer is not recognized")
+		}
+	}
+
+	customTrustRoots, err := os.ReadFile(v.TrustedRoot)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read file %s: %v", v.TrustedRoot, err)
+	}
+
+	reader := bufio.NewReader(bytes.NewReader(customTrustRoots))
+	var line []byte
+	var readError error
+	line, readError = reader.ReadBytes('\n')
+	for readError == nil {
+		// Load each trusted root
+		trustedRoot, err := root.NewTrustedRootFromJSON(line)
 		if err != nil {
-			return nil, "", fmt.Errorf("unable to read file %s: %v", v.config.TrustedRoot, err)
+			return nil, fmt.Errorf("failed to create custom verifier: %v", err)
 		}
 
-		reader := bufio.NewReader(bytes.NewReader(customTrustRoots))
-		var line []byte
-		var readError error
-		line, readError = reader.ReadBytes('\n')
-		for readError == nil {
-			// Load each trusted root
-			trustedRoot, err := root.NewTrustedRootFromJSON(line)
+		// Compare bundle leafCert issuer with trusted root cert authority
+		certAuthorities := trustedRoot.FulcioCertificateAuthorities()
+		for _, certAuthority := range certAuthorities {
+			lowestCert, err := getLowestCertInChain(&certAuthority)
 			if err != nil {
-				return nil, "", fmt.Errorf("failed to create custom verifier: %v", err)
+				return nil, err
 			}
 
-			// Compare bundle leafCert issuer with trusted root cert authority
-			certAuthorities := trustedRoot.FulcioCertificateAuthorities()
-			for _, certAuthority := range certAuthorities {
-				lowestCert, err := getLowestCertInChain(&certAuthority)
-				if err != nil {
-					return nil, "", err
-				}
-
-				if len(lowestCert.Issuer.Organization) == 0 {
-					continue
-				}
+			// if the custom trusted root issuer is not set or doesn't match the given issuer, skip it
+			if len(lowestCert.Issuer.Organization) == 0 || lowestCert.Issuer.Organization[0] != issuer {
```

Comment:
I flipped the logic checking if the cert's issuer matches the incoming issuer for an early exit in a given loop iteration.